python_bindings/setup.py

```diff
@@ -79,18 +79,19 @@ class BuildExt(build_ext):
     """A custom build extension for adding compiler-specific options."""
     c_opts = {
         'msvc': ['/EHsc', '/openmp', '/O2'],
-        'unix': ['-O3'],
+        'unix': ['-O3', '-msse4.2'],
```

Comment:
It does! However it does somewhat limit the portability of the produced binaries - I once had someone file an issue against a different project I was working on:  https://github.com/benfred/implicit/issues/16 because of this. 

However, I'm still all good with enabling -march=native for osx/linux =)